data/provider/src/main/java/tech/pegasys/teku/api/blockselector/BlockSelectorFactory.java

```diff
@@ -29,49 +29,27 @@
 import tech.pegasys.teku.storage.client.ChainHead;
 import tech.pegasys.teku.storage.client.CombinedChainDataClient;
 
-public class BlockSelectorFactory {
+public class BlockSelectorFactory extends AbstractSelectorFactory<BlockSelector> {
 
   private final Spec spec;
-  private final CombinedChainDataClient client;
 
-  public BlockSelectorFactory(
-      final Spec spec, final CombinedChainDataClient combinedChainDataClient) {
+  public BlockSelectorFactory(final Spec spec, final CombinedChainDataClient client) {
+    super(client);
     this.spec = spec;
-    this.client = combinedChainDataClient;
-  }
-
-  /**
-   * Default parsing of the slot parameter to determine the block to return - "head" - the head
-   * block - "genesis" - the genesis block - "finalized" - the block in effect at the last slot of
-   * the finalized epoch - 0x00 - the block root (bytes32) to return - {UINT64} - a specific slot to
-   * retrieve a block from
-   *
-   * @param selectorMethod the selector from the rest api call
-   * @return the selector for the requested string
-   */
-  public BlockSelector defaultBlockSelector(final String selectorMethod) {
-    if (selectorMethod.startsWith("0x")) {
-      try {
-        return forBlockRoot(Bytes32.fromHexString(selectorMethod));
-      } catch (IllegalArgumentException e) {
-        throw new BadRequestException("Invalid block: " + selectorMethod);
-      }
-    }
-    switch (selectorMethod) {
-      case "head":
-        return headSelector();
-      case "genesis":
-        return genesisSelector();
-      case "finalized":
-        return finalizedSelector();
-    }
-    try {
-      return forSlot(UInt64.valueOf(selectorMethod));
-    } catch (NumberFormatException ex) {
-      throw new BadRequestException("Invalid block: " + selectorMethod);
-    }
   }
 
+  @Override
+  public BlockSelector stateRootSelector(final Bytes32 stateRoot) {
+    throw new UnsupportedOperationException();
```

Comment:
as mentioned in DM, this will cause a 500 instead of a 400 status code we return without the change